python/python/lancedb/query.py

```diff
@@ -590,32 +590,44 @@ def to_arrow(self) -> pa.Table:
         output_tbl = self._table.to_lance().take(row_ids, columns=self._columns)
         output_tbl = output_tbl.append_column("score", scores)
 
+        if self._with_row_id:
+            # Need to set this to uint explicitly as vector results are in uint64
+            row_ids = pa.array(row_ids, type=pa.uint64())
+            output_tbl = output_tbl.append_column("_rowid", row_ids)
+
         if self._where is not None:
             try:
                 # TODO would be great to have Substrait generate pyarrow compute
                 # expressions or conversely have pyarrow support SQL expressions
                 # using Substrait
                 import duckdb
 
+                schema = output_tbl.schema
                 output_tbl = (
                     duckdb.sql("SELECT * FROM output_tbl")
                     .filter(self._where)
                     .to_arrow_table()
                 )
+                # output_tbl's fixed_size_list is now mutated to list
+                output_tbl = output_tbl.cast(schema)
```

Comment:
I'm not sure if this is the safest way to cast schema for pyarrow. I'm not sure about the implications of passing `safe=False` 
(https://arrow.apache.org/docs/1.0/python/generated/pyarrow.Table.html#pyarrow.Table.cast)